distribution/test/itests/test-itests-common/src/test/java/ddf/common/test/mock/csw/FederatedCswMockServer.java

```diff
@@ -0,0 +1,207 @@
+ * <p/>
+ * This is free software: you can redistribute it and/or modify it under the terms of the GNU Lesser
+ * General Public License as published by the Free Software Foundation, either version 3 of the
+ * License, or any later version.
+ * <p/>
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without
+ * even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details. A copy of the GNU Lesser General Public License
+ * is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package ddf.common.test.mock.csw;
+
+import static org.junit.Assert.fail;
+import static com.xebialabs.restito.semantics.Action.bytesContent;
+import static com.xebialabs.restito.semantics.Action.contentType;
+import static com.xebialabs.restito.semantics.Action.ok;
+import static com.xebialabs.restito.semantics.Condition.get;
+import static com.xebialabs.restito.semantics.Condition.post;
+import static com.xebialabs.restito.semantics.Condition.withPostBodyContaining;
+
+import java.io.IOException;
+import java.io.InputStream;
+
+import org.apache.commons.io.IOUtils;
+import org.slf4j.LoggerFactory;
+import org.slf4j.ext.XLogger;
+
+import com.xebialabs.restito.builder.stub.StubHttp;
+import com.xebialabs.restito.builder.verify.VerifyHttp;
+import com.xebialabs.restito.server.StubServer;
+
+/**
+ * Encompasses a Restito CSW Stub Server for use in integration testing. Handles handshake responses
+ * and federated query responses for a csw endpoint.
+ */
+public class FederatedCswMockServer {
+    private static final XLogger LOGGER =
+            new XLogger(LoggerFactory.getLogger(FederatedCswMockServer.class));
+
+    private final String sourceId;
+
+    private final String httpRoot;
+
+    private final int port;
+
+    private StubServer cswStubServer;
+
+    /**
+     * Constructor for the federated CSW Restito stub server.
+     *
+     * @param sourceId DDF sourceId to give to the stub server.
+     * @param httpRoot httpRoot address. Typically the non-secure root "http://localhost:".
+     * @param port     Port number on which to respond.
+     */
+    public FederatedCswMockServer(String sourceId, String httpRoot, int port) {
+        if (httpRoot.endsWith(":")) {
+            this.httpRoot = httpRoot.substring(0, httpRoot.length() - 1);
+        } else {
+            this.httpRoot = httpRoot;
+        }
+
+        this.sourceId = sourceId;
+        this.port = port;
+    }
+
+    /**
+     * Starts the Restito stub server.
+     */
+    public void start() {
+        try {
+            cswStubServer = new StubServer(port).run();
+
+            setDefaultCapabilityResponse();
+            setDefaultQueryResponse();
+        } catch (IOException | RuntimeException e) {
+            fail("Failed to setup " + getClass().getSimpleName() + ": " + e.getMessage());
```

Comment:
I don't usually care about the small stuff, but @lessarderic is this better off with `String.format`?